packages/blaze-query/src/types.ts

```diff
@@ -106,3 +108,16 @@ export abstract class Provider {
     additionalUtxos: TransactionUnspentOutput[],
   ): Promise<Redeemers>;
 }
+
+/**
+ * Mapping of RedeemerPurpose to RedeemerTag.
+ * Ensures consistency between purpose strings and tag numbers.
+ */
+export const purposeToTag: { [key: string]: number } = {
+  [RedeemerPurpose.spend]: RedeemerTag.Spend,
+  [RedeemerPurpose.mint]: RedeemerTag.Mint,
+  [RedeemerPurpose.certificate]: RedeemerTag.Cert,
+  [RedeemerPurpose.withdrawal]: RedeemerTag.Reward,
+  [RedeemerPurpose.vote]: RedeemerTag.Voting,
+  [RedeemerPurpose.propose]: RedeemerTag.Proposing,
+};
```

Comment:
Maybe worth looking into consolidating [with this function](https://github.com/butaneprotocol/blaze-cardano/blob/main/packages/blaze-query/src/blockfrost.ts#L584-L602)